azext_iot/common/utility.py

```diff
@@ -146,6 +150,44 @@ def process_json_arg(content: str, argument_name: str = "content", preserve_orde
         )
 
 
+def process_yaml_arg(path: str) -> dict:
+    """Primary processor of yaml file input"""
+
+    if not os.path.exists(path):
+        raise FileOperationError(
+            f"YAML file not found - Please ensure the path '{path}' is correct."
+        )
+
+    try:
+        import yaml
+
+        with open(path, "rb") as f:
+            return yaml.load(f, Loader=yaml.SafeLoader)
+    except Exception as ex:
+        raise InvalidArgumentValueError(
+            "Failed to parse yaml for file '{}' with exception:\n{}".format(path, ex)
+        )
+
+
+def process_toml_content(path: str) -> Dict[str, Any]:
+    """Primary processor of TOML file input"""
+
+    if not os.path.exists(path):
+        raise FileOperationError(
+            f"TOML file not found - Please ensure the path '{path}' is correct."
+        )
+
+    try:
+        import toml
```

Comment:
Where is the `toml` dependency coming from? If from az cli which part of it?

Reply:
I could have sworn it was coming from azure-cli but I think I just realized it's coming from `azdev` - might need to add this to setup / etc

Reply:
After some brief searching, I may look into `tomli/tomli-w` as well, I was only using `toml` since I figured it was already in our dependency list

Reply:
Sounds good `tomli[-w]` looks cool